### ISSUE TITLE
Operator missing permission to list nodes

### ISSUE THREAD

**Issue by clive-jevons** [emqx/emqx-operator#843]
**Describe the bug**
We have the operator deployed in version 2.1.4 and the `EMQX` cluster we provisioned never gets ready but instead is stuck in `CoreNodesUpdating`. Looking at the kubernetes events in the namespace of the CR we see:

```
LAST SEEN   TYPE      REASON                    OBJECT          MESSAGE
4m26s       Warning   FailedToGetNodeStatuses   emqx/mcs-emqx   failed to get API api/v5/nodes: failed to request API: api/v5/nodes, status: 401 Unauthorized
```

I checked the code of tag `2.1.4` and in `controllers/apps/v2alpha1/api.go` on line 120 I found:

```
func (r *requestAPI) getNodeStatuesByAPI(obj client.Object) ([]appsv2alpha1.EMQXNode, error) {
	resp, body, err := r.requestAPI(obj, "GET", "api/v5/nodes", nil)
```

When an error is returned from there, then the event is created on the EMQX resource.

**To Reproduce**
Deploy operator to a k8s cluster and provision an EMQX resource.

**Expected behavior**
The cluster is able to transition to a ready state.

**Anything else we need to know?**:

**Environment details:**:
- Kubernetes version: v1.24.10 
- Cloud-provider/provisioner: Azure AKS
- emqx-operator version: 2.1.4
- Install method: helm / ArgoCD

**Comment by Rory-Z** [comment]
Hi, @clive-jevons the `nodes` it's not Kubernetes node resource, it's EMQX's API.

**Comment by clive-jevons** [comment]
Hi @Rory-Z 

LOL - yeah, I just grokked that myself 😂🤦🏻‍♂️

This is what our CR looks like:

```
apiVersion: apps.emqx.io/v2alpha1
kind: EMQX
metadata:
  name: my-emqx
spec:
  image: emqx:5.1
```

So yeah, looks like it could be this 😬 Should I try setting the image to `emqx:5.0.23`? WDYT?

**Comment by Rory-Z** [comment]
Yes, I think the `emqx:5.0.23` will worked

**Comment by clive-jevons** [comment]
@Rory-Z I can confirm that everything worked with image set to `emqx:5.0.23` 😁👍👍

WDYT - shall I leave this issue open until it's fixed for `>5.0.23`?

**Comment by Rory-Z** [comment]
> shall I leave this issue open until it's fixed for >5.0.23?

All right, please look forward out work

**Comment by clive-jevons** [comment]
@Rory-Z  - Thank you for your help and quick response 😍👍

**Comment by erezblm** [comment]
hey @clive-jevons , any expected time for the fix to be released? I have the same issue here with EMQX 5.1.X, and we really want to work with this EMQX version.

IS there a workaround to get behind the issue for now? deploying the RBAC manually or something?

Thanks a lot

**Comment by Rory-Z** [comment]
Hi @erezblm Operator 2.2.0 will coming soon.
We already released 2.2.0-beta.3, you can try it

**Comment by Rory-Z** [comment]
EMQX Operator 2.2.0 is released, close this issue
